src/Microsoft.ApplicationInsights.AspNetCore/Logging/Implementation/ApplicationInsightsLoggerFactoryExtensions.cs

```diff
@@ -67,26 +67,30 @@ public static ILoggerFactory AddApplicationInsights(this ILoggerFactory factory,
             Func<string, LogLevel, bool> filter,
             Action loggerAddedCallback)
         {
-            var client = serviceProvider.GetService<TelemetryClient>();
-            var debugLoggerControl = serviceProvider.GetService<ApplicationInsightsLoggerEvents>();
-            var options = serviceProvider.GetService<IOptions<ApplicationInsightsLoggerOptions>>();
-
-            if (options == null)
+            var loggerMarker = serviceProvider.GetService<LoggerMarker>();
```

Comment:
Instead of doing this, maybe we can do IStartupFilter?

Reply:
Essentially above in #If NetStandard2.0 we can put an additional #else and add a startup filter for starting up the old path.

Reply:
@RamjotSingh Is there any scenario broken with this? Or just a suggestion?

Reply:
Probably not broken (not sure). But this code looked like it was abusing a DI construct. Hence the suggestion.

Reply:
This need revisit. By doing this, i am essentially breaking behavior already - i.e users already using logerFactory.AddApplicationInsights will see that their Filters are not respected, as this PR now favors the new Provider.

Need to do the inverse somehow - i.e for existing loggerFactory.AddApplicationInsights users, the new Provider should be disabled.

Reply:
We don't need to detect this actually. Since the old ILoggerFactory method is marked obsolete, we expect customers to stop using it. I will improve the Obsolete warning to be more descriptive and link to official docs.

Reply:
closing this as not longer applicable.